Test failure with cowsay installed/present

### Summary

When running the build/tests for "test_broken_cowsay" the test fails if the cowsay package is present/available on the system:

```
=================================== FAILURES ===================================
_____________________ test_display_with_fake_cowsay_binary _____________________
[gw13] linux -- Python 3.12.3 /usr/bin/python

capsys = <_pytest.capture.CaptureFixture object at 0x75e7f5486840>
mocker = <pytest_mock.plugin.MockerFixture object at 0x75e7f5485f70>

    def test_display_with_fake_cowsay_binary(capsys, mocker):
        display = Display()

        mocker.patch("ansible.constants.ANSIBLE_COW_PATH", "./cowsay.sh")

        mock_popen = MagicMock()
        mock_popen.return_value.returncode = 1
        mocker.patch("subprocess.Popen", mock_popen)

>       assert not hasattr(display, "cows_available")
E       AssertionError: assert not True
E        +  where True = hasattr(<ansible.utils.display.Display object at 0x75e7f7fb16a0>, 'cows_available')

test/units/utils/display/test_broken_cowsay.py:23: AssertionError
```

### Issue Type

Bug Report

### Component Name

test/units/utils/display/test_broken_cowsay.py

### Ansible Version

```console
$ ansible --version
ansible [core 2.16.7]
  config file = /home/ssilverman/.ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.12/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.12.3 (main, Apr 23 2024, 09:16:07) [GCC 13.2.1 20240417] (/usr/bin/python)
  jinja version = 3.1.3
  libyaml = True
```


### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
ANSIBLE_NOCOWS(/home/user/.ansible.cfg) = True
ANSIBLE_PIPELINING(/home/user/.ansible.cfg) = True
CONFIG_FILE() = /home/user/.ansible.cfg
DEFAULT_FORKS(/home/user/.ansible.cfg) = 30
DEFAULT_HOST_LIST(/home/user/.ansible.cfg) = ['/home/user/ansible/inventory']
DEFAULT_LOG_PATH(/home/user/.ansible.cfg) = /home/user/ansible/ansible.log
DISPLAY_ARGS_TO_STDOUT(/home/user/.ansible.cfg) = True
EDITOR(env: EDITOR) = vim
INTERPRETER_PYTHON(/home/user/.ansible.cfg) = auto_silent
RETRY_FILES_ENABLED(/home/user/.ansible.cfg) = True
RETRY_FILES_SAVE_PATH(/home/user/.ansible.cfg) = /home/user/ansible/retry

CONNECTION:
==========

local:
_____
pipelining(/home/user/.ansible.cfg) = True

psrp:
____
pipelining(/home/user/.ansible.cfg) = True

ssh:
___
pipelining(/home/user/.ansible.cfg) = True

winrm:
_____
pipelining(/home/user/.ansible.cfg) = True
```


### OS / Environment

Arch Linux

### Steps to Reproduce

<!--- Paste example playbooks or commands between quotes below -->
```yaml (paste below)
Issue with tests/build, not executing playbooks
```


### Expected Results

I expected the tests to pass, whether or not cowsay was present.

### Actual Results

```console
=================================== FAILURES ===================================
_____________________ test_display_with_fake_cowsay_binary _____________________
[gw13] linux -- Python 3.12.3 /usr/bin/python

capsys = <_pytest.capture.CaptureFixture object at 0x75e7f5486840>
mocker = <pytest_mock.plugin.MockerFixture object at 0x75e7f5485f70>

    def test_display_with_fake_cowsay_binary(capsys, mocker):
        display = Display()

        mocker.patch("ansible.constants.ANSIBLE_COW_PATH", "./cowsay.sh")

        mock_popen = MagicMock()
        mock_popen.return_value.returncode = 1
        mocker.patch("subprocess.Popen", mock_popen)

>       assert not hasattr(display, "cows_available")
E       AssertionError: assert not True
E        +  where True = hasattr(<ansible.utils.display.Display object at 0x75e7f7fb16a0>, 'cows_available')

test/units/utils/display/test_broken_cowsay.py:23: AssertionError
```


### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

--- a/changelogs/fragments/83327.yml
+++ b/changelogs/fragments/83327.yml
@@ -1,0 +1,3 @@
+---
+bugfixes:
+  - fixed unit test test_borken_cowsay to address mock not been properly applied when existing unix system already have cowsay installed.

--- a/test/units/utils/display/test_broken_cowsay.py
+++ b/test/units/utils/display/test_broken_cowsay.py
@@ -10,13 +10,12 @@ from unittest.mock import MagicMock
 
 
 def test_display_with_fake_cowsay_binary(capsys, mocker):
-    display = Display()
 
     mocker.patch("ansible.constants.ANSIBLE_COW_PATH", "./cowsay.sh")
-
     mock_popen = MagicMock()
     mock_popen.return_value.returncode = 1
     mocker.patch("subprocess.Popen", mock_popen)
+    display = Display()
 
     assert not hasattr(display, "cows_available")
     assert display.b_cowsay is None

